rest/stats_context.go

```diff
@@ -320,19 +323,20 @@ func discoverInterfaceName(host string) (interfaceName string, err error) {
 		if err != nil {
 			return "", err
 		}
+
 		for _, ifaceCIDRAddr := range ifaceAddresses {
 
 			ipAddr, _, err := net.ParseCIDR(ifaceCIDRAddr.String())
 			if err != nil {
 				return "", err
 			}
 
-			if ipAddr.String() == host {
+			if _, ok := hosts[ipAddr.String()]; ok {
```

Comment:
In the scenario where a host has multiple addresses, is it possible for those addresses to assigned to different interfaces?

Reply:
Yes it's possible, but the interface _actually_ being used for the REST API depends what was first in the resolved list of addresses at the time of `net.Listen()` [which does not support multiple addresses](https://github.com/golang/go/issues/9334).

I think that means that as long as `LookupHost` resolves in the same order as `net.Listen`, you'll get correct results - but if that does not happen, you'll see stats for another interface. I don't know how likely or unlikely this is.

There's a "correct" implementation that resolves an address _before_ starting up the HTTP servers, so we know exactly what to look for in this stat code, but I'm not sure it's worth that to cover this specific case.

Reply:
Makes sense - thanks.